distributed/utils.py

```diff
@@ -1712,3 +1713,18 @@ def is_python_shutting_down() -> bool:
     from distributed import _python_shutting_down
 
     return _python_shutting_down
+
+
+_CoroutineFunctionType = TypeVar(
+    "_CoroutineFunctionType", bound=Callable[..., Coroutine]
+)
+
+
+def delayed(corofunc: _CoroutineFunctionType, delay: float) -> _CoroutineFunctionType:
+    """Decorator to delay the evaluation of a coroutine function by the given delay in seconds."""
+
+    async def wrapper(*args, **kwargs):
+        await asyncio.sleep(delay)
+        return await corofunc(*args, **kwargs)
+
+    return cast(_CoroutineFunctionType, wrapper)
```

Comment:
does this work with a paramspec instead of a cast?

```python
_P = ParamSpec("_P")
_T = TypeVar("_T")

def delayed(corofunc: Callable[_P, Coroutine[Any, Any, _T], delay: float) -> Callable[_P, Coroutine[Any, Any, T]]:
    """Decorator to delay the evaluation of a coroutine function by the given delay in seconds."""

    async def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> T:
        await asyncio.sleep(delay)
        return await corofunc(*args, **kwargs)

    return wrapper
```

Reply:
Looks like it does, thanks for pointing me toward ParamSpec.